### DISCUSSION TITLE
Support Layer7 network policies

### DISCUSSION THREAD

**Discussion by ferrandinand** [aws/amazon-vpc-cni-k8s#2759] [Q&A]
We are adopting vpc-cni as our stantdard cni in our clusters and using it also to work with our network policies but we have been  required now to filter traffic by DNS hostnames with L7 rules.
This is possible with Calico Enterprise and Cillium but seems an architectural challenge in vpc-cni.

Cillium also has a long open issue to work along with vpc-cni L7 with many challenges.
https://github.com/cilium/cilium/issues/12454

Is there any possibility to support this use case in the future? is this architectural possible?

**Comment by jdn5126** [comment]
@ferrandinand I suggest filing a Containers Roadmap request here: https://github.com/aws/containers-roadmap/issues to get the best visibility from Product Managers
